### DISCUSSION TITLE
When are .dot files created

### DISCUSSION THREAD

**Discussion by tgfrerer** [tgfrerer/island#20] [Q&A] [answered]
via: https://github.com/tgfrerer/island/issues/18#issue-788552752

> While working on le_video, I noticed that a new .dot file is generated in every frame and spamming the output. is this the intended behaviour or did I somehow change the internal system so it thinks it needs to recreate the viz?

**Comment by tgfrerer** [comment]
# What are .dot files?
![graphviz-screenshot](https://raw.githubusercontent.com/tgfrerer/island/wip/resources/readme/graph_screenshot.png)

.dot files contain a visual representation of the current rendergraph. You can convert a .dot file into a PDF using the dot command line utility on a linux machine. There is also a very convenient plug-in for Visual Studio Code which allows you to preview a .dot file directly.

# When are .dot files generated?
.dot files are generated for Debug applications by default as soon as the le_renderer detects a change in the Rendergraph. 
In some cases, this can lead to `.dot` files being generated for every frame, e.g. in case there is a ping-pong scenario where the rendergraph oscillates between two (or more) states.

# How disable .dot file generation
If you don't want to generate .dot files for a particular app you can add the following line to the app's `CMakeLists.txt` file (meaning the CMake file which is in the same directory as your app's `main.cpp`):

```
add_compile_definitions(DEBUG_GENERATE_DOT_GRAPH=false)
```
This should sit just above the line setting your application PROJECT_NAME.

**Reply by underdoeg** [reply]
FYI on linux you can also use graphUI to view the plot files https://github.com/artemanufrij/graphui

**Comment by tgfrerer** [comment, marked as answer]
Update: as of 0.7.0, dot files are generated whenever the associated LE_SETTING is > 0. 

* For queue sync .dot files, the LE_SETTING is: `LE_SETTING_GENERATE_QUEUE_SYNC_DOT_FILES`
* For rendergraph .dot filer, the LE_SETTING is: `LE_SETTING_RENDERGRAPH_GENERATE_DOT_FILES`

Whenever a .dot file is generated, the corresponding LE_SETTING is decremented by the backend.

To tell the backend to create a .dot file, do this (for example as a response to a key press): 
```
LE_SETTING( uint32_t, LE_SETTING_GENERATE_QUEUE_SYNC_DOT_FILES, 0 ); // make setting available	
*LE_SETTING_GENERATE_QUEUE_SYNC_DOT_FILES = 1; // set setting to 1

```
You can set `LE_SETTING_GENERATE_QUEUE_SYNC_DOT_FILES` to `10` if you want to create .dot files for the next 10 frames.
